Save settings immediately upon changing in RunActivity.exe

YoRyan opened the pull request:
Changes the code to save whitelisted settings immediately upon change. Also changes the code to respect the skip-user-settings command-line flag.

Bug report: https://github.com/openrails/openrails/pull/306#issuecomment-803592877

twpol commented on Source/ORTS.Common/SettingsBase.cs:
```diff
@@ -43,7 +43,15 @@ protected enum Source
+        /// <summary>
+        /// Translates name of a setting to its value when first loaded.
+        /// </summary>
+        protected readonly IDictionary<string, object> InitialValues = new Dictionary<string, object>();
```
The save system (`SettingsBase.Save` IIRC) already handles the prospect of saving a setting with a value equal to its default (it removes it), so this `InitialValues` might be redundant?

twpol commented on Source/RunActivity/Program.cs:
```diff
@@ -65,7 +65,7 @@ static void Main(string[] args)
             game.Run();
 
-            settings.Save();
+            settings.SaveInGameChanges();
```
An objection to this whole approach is that saving on exit requires the program to exit cleanly, including the computer not crashing.

And targeting these specific settings is contrary to what the existing code does, which is explicitly saving when things are changed: https://github.com/openrails/openrails/blob/69600d7daa88e3340bebe894aaa0f226b81d3495/Source/RunActivity/Viewer3D/Popups/Window.cs#L105 (granted that also doesn't respect `/skip-user-settings` but that would be an easy fix too)

While this PR improves the behaviour of the newly added saving of settings, it is still disjoint from the pre-existing method and I would like to see that addressed somewhere.

YoRyan committed:
save in-game settings immediately upon changing them

Creates a new SavingProperty class to wrap properties that should be saved upon change.

YoRyan committed:
respect the skip-user-settings flag when changing settings

YoRyan committed:
remove save operation on entire settings store

YoRyan committed:
fix formatting

YoRyan committed:
use the last selected mode for OSD locations

YoRyan commented:
Thanks for your feedback, James. I've adopted the save-immediately-upon-change approach you suggested. To accomplish this, I created a new class; I do wish I had a niftier way, but the C# typing system makes it difficult to get fancy.

twpol reviewed (approved):
Thanks very much, this is great!

YoRyan commented:
Thanks! I'll give it a week in unstable.
